databroker/_core.py

```diff
@@ -2156,8 +2167,17 @@ def from_config(cls, config, auto_register=True, name=None):
         # Instantiate component classes.
         mds = mds_cls(config['metadatastore']['config'])
         assets = assets_cls(config['assets']['config'])
+        # Instantiate event sources.
+        event_sources = []
+        ess = []
+        if 'event_sources' in config.keys():
+            ess = config['event_sources']
+        for es in ess:
+            es_cls = load_cls(es)
+            event_sources.append(es_cls._from_config(es['config']))
```

Comment:
I think this can just be `es_cls(es['config'])` and the `ArchiverEventSource.from_config` method can be removed. The reason that `Broker` has `from_config` is that it can be handed live `mds` and `reg` objects (via `__init__`) or configuration telling it how to go build them (via `from_config`). In this situation, I think it is fine (and obviously simpler) to only support the `__init__` path.